nucliadb/src/nucliadb/ingest/orm/processor/data_augmentation.py

```diff
@@ -0,0 +1,159 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+import logging
+from dataclasses import dataclass, field
+from typing import Optional
+
+from nucliadb.ingest.orm.resource import Resource
+from nucliadb.ingest.processing import ProcessingEngine, PushPayload, Source
+from nucliadb_models.text import PushTextFormat, Text
+from nucliadb_protos import resources_pb2, writer_pb2
+from nucliadb_protos.resources_pb2 import FieldType
+from nucliadb_utils.utilities import Utility, get_partitioning, get_utility
+
+logger = logging.getLogger("ingest-processor")
+
+
+@dataclass
+class GeneratedFields:
+    texts: list[str] = field(default_factory=list)
+    links: list[str] = field(default_factory=list)
+    files: list[str] = field(default_factory=list)
+    conversations: list[str] = field(default_factory=list)
+
+    def __len__(self):
```

Comment:
This is cool, although it made me do a double take (I saw the `len(generated_fields)` first and I was surprised classes in Python had length :sweat_smile:).

Anyway, since we only care if there are generated fields or no, what about replacing this function for something like `is_empty()`. It would express our intent more explicitly.